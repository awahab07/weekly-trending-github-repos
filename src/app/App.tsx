import { CssBaseline, ThemeProvider } from '@material-ui/core';
import React from 'react';
import { BrowserRouter as Router, Redirect, Route, Switch } from 'react-router-dom';
import RepositoryContainer from '../features/repository/components/RepositoryContainer';
import { BoxGrid } from '../shared/BoxComponents';
import { appTheme, useAppStyles } from './theme';

const App: React.FunctionComponent = (): React.ReactElement => {
  const classes = useAppStyles();

  return (
    <ThemeProvider theme={appTheme}>
      <CssBaseline>
        <Router>
          <BoxGrid container classes={{ root: classes.rootWrapper }} justify={'center'}>
            <BoxGrid className={classes.contentWrapper}>
              <Switch>
                <Route path="/repositories">
                  <RepositoryContainer />
                </Route>
                <Redirect exact={true} from="/" to="/repositories" />
              </Switch>
            </BoxGrid>
          </BoxGrid>
        </Router>
      </CssBaseline>
    </ThemeProvider>
  );
};

export default App;
