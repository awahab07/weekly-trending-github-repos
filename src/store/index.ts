export { default, acceptHotStore } from './store';
export * from './root.reducer';
