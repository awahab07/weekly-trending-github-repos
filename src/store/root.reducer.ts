import { combineReducers } from '@reduxjs/toolkit';

import { repositoryReducer } from '../features/repository';

const rootReducer = combineReducers({
  repositorySlice: repositoryReducer,
});

export type RootState = ReturnType<typeof rootReducer>;

export default rootReducer;
