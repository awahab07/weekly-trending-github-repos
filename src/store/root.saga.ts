import { all, fork } from 'redux-saga/effects';
import { repositoryListener } from '../features/repository';

export const rootSaga = function* root(): Generator {
  yield all([fork(repositoryListener)]);
};
