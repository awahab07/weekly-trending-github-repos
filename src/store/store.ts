import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import createSagaMiddleware from 'redux-saga';
import { HmrAcceptor } from '../app/hmr';

import rootReducer from './root.reducer';
import { rootSaga } from './root.saga';

const sagaMiddleware = createSagaMiddleware();

const middleware = [...getDefaultMiddleware({ thunk: false }), sagaMiddleware];

const store = configureStore({
  reducer: rootReducer,
  middleware,
});

sagaMiddleware.run(rootSaga);

export const acceptHotStore = (accept: HmrAcceptor): void => {
  accept('./root.reducer', (): void => {
    const newRootReducer = require('./root.reducer').default;
    store.replaceReducer(newRootReducer);
  });
};

export default store;
