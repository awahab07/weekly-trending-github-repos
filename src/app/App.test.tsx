import { shallow } from 'enzyme';
import React from 'react';
import { StaticRouter } from 'react-router';
import { MockStoreProvider } from '../store/test';
import App from './App';

declare const global: any;

global.window = Object.create(window);
const url = 'http://test.com/category/product';
Object.defineProperty(window, 'location', {
  value: {
    href: url
  }
});

describe('App renders correctly', () => {
  it('should render correctly', () => {
    const Component = (
      <MockStoreProvider>
        <StaticRouter>
          <App />
        </StaticRouter>
      </MockStoreProvider>
    );
    const component = shallow(Component);

    expect(component).toMatchSnapshot();
  });
});
