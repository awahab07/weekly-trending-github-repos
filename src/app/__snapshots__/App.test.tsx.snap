// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly should render correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <StaticRouter>
    <App />
  </StaticRouter>
</Provider>
`;
